src/Sulu/Bundle/MediaBundle/Media/Manager/MediaManager.php

```diff
@@ -906,6 +915,42 @@ public function addFormatsAndUrl(Media $media)
             $media->setUrl($versionData[$media->getVersion()]['url']);
         }
 
+        // Get preview image and overwrite thumbnails if set
+        /** @var \Sulu\Bundle\MediaBundle\Entity\Media $previewImage */
+        $previewImage = $media->getEntity()->getPreviewImage();
+
+        if ($previewImage !== null) {
+            /** @var FileVersion $latestVersion */
+            $latestVersion = null;
+
+            /** @var File $file */
+            foreach ($previewImage->getFiles() as $file) {
+                foreach ($file->getFileVersions() as $version) {
+                    $highestVersion = 0;
+
+                    if ($version->getVersion() > $highestVersion) {
+                        $latestVersion = $version;
+                        $highestVersion = $version->getVersion();
+                    }
+                }
+
+                // currently only one file per media exists
+                break;
+            }
+
+            if ($latestVersion !== null) {
+                $media->setFormats(
```

Comment:
I can't imagine that we really want to override the existing formats here. How can you then access the video itself?